File: expectanim/src/main/java/com/github/florent37/expectanim/core/position/PositionAnimExpectationRightOfParent.java
package com.github.florent37.expectanim.core.position;

import android.view.View;
import android.view.ViewParent;

/**
 * Created by florentchampigny on 17/02/2017.
 */

public class PositionAnimExpectationRightOfParent extends PositionAnimExpectation {

    public PositionAnimExpectationRightOfParent() {
        setForPositionX(true);
    }

    @Override
    public Float getCalculatedValueX(View viewToMove) {
        final ViewParent viewParent = viewToMove.getParent();
        if((viewParent instanceof View)){
            final View parentView = (View)viewParent;
            return parentView.getWidth() - getMargin(viewToMove) - viewCalculator.finalWidthOfView(viewToMove);
        }
        return null;
    }

    @Override
    public Float getCalculatedValueY(View viewToMove) {
        return null;
    }
}

File: expectanim/src/main/java/com/github/florent37/expectanim/core/position/PositionAnimExpectationRightOfParent.kt
package com.github.florent37.expectanim.core.position

import android.view.View
import android.view.ViewParent

class PositionAnimExpectationRightOfParent : PositionAnimExpectation() {

    init {
        isForPositionX = true
    }

    override fun getCalculatedValueX(viewToMove: View): Float? {
        val viewParent = viewToMove.parent
        if (viewParent is View) {
            val parentView = viewParent as View
            return parentView.width.toFloat() - getMargin(viewToMove) - viewCalculator!!.finalWidthOfView(viewToMove)
        }
        return null
    }

    override fun getCalculatedValueY(viewToMove: View): Float? {
        return null
    }
}
